Display tags in runs comparison table within the UI

CaioCavalcanti opened the pull request:
<!-- 🚨 We recommend pull requests be filed from a non-master branch on a repository fork (e.g. <username>:fix-xxx). 🚨 -->

## Related Issues/PRs

<!--
Please reference any related feature requests, issues, or PRs here. For example, `#123`. To automatically close the referenced items when this PR is merged, please use a closing keyword (close, fix, or resolve). For example, `Close #123`. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for more information.
-->
Close #6082

## What changes are proposed in this pull request?

In the runs comparison table within the UI, display tags associated with runs.

![image](https://user-images.githubusercontent.com/3620566/177014272-3e7da75e-4982-4d3f-8a90-c4573bd8a06c.png)

## How is this patch tested?

<!--
If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask. We'd be happy to help.
-->
- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

https://user-images.githubusercontent.com/3620566/177013742-dc02941a-b175-455a-bf1b-353340bfb55c.mp4


<!--
Please describe how you confirmed the proposed feature/bug-fix/change works here. For example, if you fixed an MLflow client API, you could attach the code that didn't work prior to the fix but works now, or if you added a new feature on MLflow UI, you could attach a video that demonstrates the feature.
-->

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Show a table of run tags when comparing experiment runs.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->
<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

CaioCavalcanti committed:
add tags to run comparison table

Signed-off-by: Caio Cavalcanti <caiofabiomc@gmail.com>

CaioCavalcanti committed:
Merge remote-tracking branch 'origin/master' into fr-tags-on-run-comparison-table

Signed-off-by: Caio Cavalcanti <caiofabiomc@gmail.com>

CaioCavalcanti commented:
- [ ] Attach a video that demonstrates the feature

harupy commented:
> * [ ]  Attach a video that demonstrates the feature

Awesome

harupy commented on mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:
```diff
@@ -261,6 +262,34 @@ export class CompareRunView extends Component {
+    return (
+      <table
+        className='table compare-table compare-run-table'
+        style={{ maxHeight: '500px' }}
```
```suggestion
        css={{ maxHeight: '500px' }}
```
```suggestion
        style={{ maxHeight: '500px' }}
```

CaioCavalcanti replied:
`css` is also used in other places in this file. Can I fix all in the same change?

harupy replied:
Yes!

CaioCavalcanti replied:
Done, fixed all of them.

harupy commented on mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:
```diff
@@ -261,6 +262,34 @@ export class CompareRunView extends Component {
+
+  renderTagTable(colWidth) {
+    const dataRows = this.renderDataRows(
+      this.props.tagLists,
```
@CaioCavalcanti We discussed whether we should show MLflow system tags today. Here's our conclusion:

> If we show system tags on the run comparison page, we should probably show them on all pages (e.g. Runs table, Run Page, etc) for consistency. However, the Run Page lets you edit tags, yet system tags should not be edited arbitrarily. This could create confusion and buggy behavior in features that depend on particular system tag values. For this reason, I think it's best not to surface system tags in the UI and instead encourage users to call the API if they're interested in viewing these tags.

harupy replied:
Function to filter out MLflow system tags:

https://github.com/mlflow/mlflow/blob/4c0f21025322f4681e5c4cf01d2ad57e0eb93115/mlflow/server/js/src/common/utils/Utils.js#L798-L805

CaioCavalcanti replied:
Done, now it shows only the visible tags.

harupy commented:
@CaioCavalcanti To format JS code, you can comment `autoformat`.

harupy commented:
@CaioCavalcanti Any updates here?

CaioCavalcanti commented:
> @CaioCavalcanti Any updates here?

@harupy I got stuck with an unrelated issue on my env that's blocking me from testing the changes (node stopping due to out of memory in a container on WSL).

I'll take a look later today.

CaioCavalcanti committed:
Show only visible tags on compare runs

Also fix lint issues and replace `style` attribute for `css` on html
elements.

Signed-off-by: Caio Cavalcanti <caiofabiomc@gmail.com>

CaioCavalcanti committed:
Update default translation

Signed-off-by: Caio Cavalcanti <caiofabiomc@gmail.com>

CaioCavalcanti committed:
Fix tests for CompareRunView

Signed-off-by: Caio Cavalcanti <caiofabiomc@gmail.com>

harupy reviewed (approved):
LGTM :)

harupy committed:
use array destructuring

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy commented on mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:
```diff
@@ -588,6 +633,12 @@ const mapStateToProps = (state, ownProps) => {
     metricLists.push(Object.values(getLatestMetrics(runUuid, state)));
     paramLists.push(Object.values(getParams(runUuid, state)));
     const runTags = getRunTags(runUuid, state);
+    const visibleTags = Utils.getVisibleTagValues(runTags).map(([key, value]) => ({
+      key,
+      name: key,
+      value,
+    }));
+    tagLists.push(visibleTags);
```
@CaioCavalcanti I pushed one commit to clean up the code here.

CaioCavalcanti replied:
Nice, good one

harupy committed:
udpate test

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
remove name

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy commented:
@CaioCavalcanti Thanks for the contribution, merged!
